src/adapter/graphql/src/queries/flows/flow_start_condition.rs

```diff
@@ -19,12 +21,47 @@ pub(crate) enum FlowStartCondition {
     Batching(FlowStartConditionBatching),
 }
 
-impl From<fs::FlowStartCondition> for FlowStartCondition {
-    fn from(value: fs::FlowStartCondition) -> Self {
-        match value {
-            fs::FlowStartCondition::Throttling(t) => Self::Throttling(t.into()),
-            fs::FlowStartCondition::Batching(b) => Self::Batching(b.into()),
-        }
+impl FlowStartCondition {
+    pub async fn create_from_raw_flow_data(
+        flow_state: &fs::FlowState,
+        dataset_changes_service: &dyn DatasetChangesService,
+    ) -> Result<Option<Self>, InternalError> {
+        Ok(match &flow_state.start_condition {
+            None => None,
+            Some(fs::FlowStartCondition::Throttling(t)) => Some(Self::Throttling((*t).into())),
+            Some(fs::FlowStartCondition::Batching(b)) => {
+                // Start from zero increment
+                let mut total_increment = DatasetIntervalIncrement::default();
+
+                // TODO: somehow limit dataset traversal to blocks that existed at the time of
+                // flow latest event, as they might have evolved after this state was loaded
+
+                // For each dataset trigger, add accumulated changes since trigger first fired
+                for trigger in &flow_state.triggers {
+                    if let fs::FlowTrigger::InputDatasetFlow(dataset_trigger) = trigger {
+                        if let fs::FlowResult::DatasetUpdate(dataset_update) =
+                            &dataset_trigger.flow_result
+                        {
+                            total_increment += dataset_changes_service
+                                .get_increment_since(
+                                    &dataset_trigger.dataset_id,
+                                    dataset_update.old_head.as_ref(),
+                                )
+                                .await
+                                .int_err()?;
+                        }
+                    }
+                }
```

Comment:
Let's suppose the metadata chain of an input looks like this:
```
... <- b1 <- b2 <- b3 <- b4 <- b5
```

Our derivative dataset's flow contains this set of triggers:
```
InputDatasetFlow(DatasetUpdate(old_head=b2, new_head=b3))
InputDatasetFlow(DatasetUpdate(old_head=b3, new_head=b4))
```

While in our last `ExecuteTransform` the last processed input block is `b1`.

i.e. we **for whatever reason missed two transitions**:
```
b1 to b2
b4 to b5
```

If I read this right - this code seems to calculate something completely whacky like:
```
num_records (b2, b5] + num_records (b3, b5]
```

While it should either do:
- `num_records (b2, b3] + num_records (b3, b4]` - sum of intervals our triggers know about
- `num_records (b2, b5]` - using only **first trigger per input** and allowing to scan until the current `head`
- `num_records (b1, b5]` - **the correct answer**, but would require finding last `ExecuteTransform` of our derivative dataset instead of looking at triggers

Reply:
When we last spoke I saw two solutions:
- Store only one trigger per dataset and calculate all batching info from metadata of both derivative dataset and its inputs (using last `ExecuteTransfrom`)
- Store all triggers with `FlowResult::DatasetUpdate` inside of them and calculate batching based on stats **inside those events**

Current solution seems in between the two - it stores all triggers, but still evaluates batching based on inputs' metadata. 

Am I misunderstanding something?

Reply:
1) We are storing 1 trigger per dataset
2) The assumption is that flow system cannot miss any update in the middle of execution. If we disablee server for a while and worked directly with the storage, it will catch the updates up during initial startup phase, where it's scheduling unconditional update for all enabled derived datasets.

Reply:
Proposed on Slack to record the final `StartCondition` that made us decide to start the flow as part of the start event.

Reply:
Decided to hold